### ISSUE TITLE
DatePicker shortcode does not work from endpoints

### ISSUE THREAD

**Issue by zackdutra** [Triumph-Tech/Triumph-Helix#11]
### Description

The datepicker shortcode works when added directly to a page, but not when loaded from an HTMX endpoint.

### Actual Behavior

If you click on the datepicker field rendered by the datepicker shortcode, nothing happens. 

### Expected Behavior

An element should appear allowing you to select a date. 

### Steps to Reproduce

- Create a test lava application with a slug of test
- Add a first endpoint as a GET, security of application view, a slug of test-date and the code template below, which is copied directly from the pre-installed lava shortcode example:
`{[ datepicker label:'Start Date' name:'start-date' isrequired:'true' value:'{{ 'Now' | Date:'M/d/yyyy' }}' ]}`

- Add a page with a lava application content block, with the code template set to the code below:
```
<div id="main" hx-get="^/test/test-date" hx-trigger="load" >
</div>

<hr>

{[ datepicker label:'Start Date' name:'start-date' isrequired:'true' value:'{{ 'Now' | Date:'M/d/yyyy' }}' ]}
```

- Save the code, reload the page, and note that there are two inputs rendered, but the one from the endpoint doesn't show a selector when you click on it: 

![Screenshot 2024-10-17 at 4 03 46 PM](https://github.com/user-attachments/assets/d81da0c6-76e0-442d-939b-ab7561c3d2f9)
![Screenshot 2024-10-17 at 4 03 41 PM](https://github.com/user-attachments/assets/ea30ff52-f60e-48da-99b9-9b53b5a6b0af)

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16.6

**Comment by zackdutra** [comment]
Adding data-provide="datepicker" to the input-group seems to resolve this, so the shortcode would be edited to the following: 
```

{% assign isrequired = isrequired | AsBoolean %}
{% assign showlabel = showlabel | AsBoolean %}

{[ rockcontrol id:'rc-{{ '' | UniqueIdentifier }}' label:'{{ label }}' showlabel:'{{ showlabel }}' controltype:'date-picker' isrequired:'{{ isrequired }}' validationmessage:'{{ validationmessage }}' ]}

    <div class="input-group input-width-md js-date-picker date" data-provide="datepicker">
    	<input name="{{ name }}" type="text" id="{{ id }}" class="form-control" value="{{ value }}" {% if isrequired == true %}required{% endif %} {{ additionalattributes }}> <span class="input-group-addon"><i class="fa fa-calendar"></i></span>
    </div>

    <script>
        document.addEventListener("DOMContentLoaded", function() {
            Rock.controls.datePicker.initialize(
                {
                    id: '{{ id }}',
                    startView: 0,
                    showOnFocus: true,
                    format: 'mm/dd/yyyy',
                    todayHighlight: true,
                    forceParse: true,
                    postbackScript: '',
                });
        });
    </script>
    
{[ endrockcontrol ]}

```

**Comment by zackdutra** [comment]
Realizing that this seems more like a JS issue, here's another potential solution in the script tag. 

```

{% assign isrequired = isrequired | AsBoolean %}
{% assign showlabel = showlabel | AsBoolean %}

{[ rockcontrol id:'rc-{{ '' | UniqueIdentifier }}' label:'{{ label }}' showlabel:'{{ showlabel }}' controltype:'date-picker' isrequired:'{{ isrequired }}' validationmessage:'{{ validationmessage }}' ]}

    <div class="input-group input-width-md js-date-picker date">
    	<input name="{{ name }}" type="text" id="{{ id }}" class="form-control" value="{{ value }}" {% if isrequired == true %}required{% endif %} {{ additionalattributes }}> <span class="input-group-addon"><i class="fa fa-calendar"></i></span>
    </div>

    <script>
        (function() {
            Rock.controls.datePicker.initialize({
                id: '{{ id }}',
                startView: 0,
                showOnFocus: true,
                format: 'mm/dd/yyyy',
                todayHighlight: true,
                forceParse: true,
                postbackScript: '',
            });
        })();
    </script>
    
{[ endrockcontrol ]}

```

**Comment by zackdutra** [comment]
For my own temporary fix, rather than using a custom lava shortcode, I'm doing this, which seems to work for me: 

 ```
   {[ datepicker label:'Start Date' name:'start-date' isrequired:'true' value:'{{ 'Now' | Date:'M/d/yyyy' }}' ]}

    <script>
        (function() {
            let datePicker = document.getElementsByName('start-date')[0];
            let datePickerId = datePicker.id;
            Rock.controls.datePicker.initialize({
                id: datePickerId,
                startView: 0,
                showOnFocus: true,
                format: 'mm/dd/yyyy',
                todayHighlight: true,
                forceParse: true,
                postbackScript: '',
            });
        })();
    </script>
```

**Comment by Panha-Sim** [comment]
@zackdutra This issue has been fixed in Helix v0.9, which is coming soon.
